<launch>
  
  <arg name="nao_ip1" default="$(optenv NAO_IP 192.168.1.19)"/>
  <arg name="nao_port1" default="$(optenv NAO_PORT 9559)"/>
  
  <!--
    Top camera 1
  -->
  <node pkg="nao_driver" type="nao_camera.py" name="nao_camera1" required="true" args="--pip=$(arg nao_ip1) --pport=$(arg nao_port1)" output="screen">
    <param name="camera_switch" value="0"/>
    <param name="resolution" value="2"/>
    <remap from="image_raw" to="image_raw1"/>
  </node>
  
  <!--
    Leds 1
  -->
  <node pkg="nao_behavior_tree" type="nao_leds" name="nao_leds" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip1)"/>
    <param name="NAO_PORT" value="$(arg nao_port1)"/>
  </node>
  
  
  <!--
    Actions 1
  -->
  <node pkg="nao_behavior_tree" type="TrackingAmongObstacles" name="TrackingAmongObstacles1" args="1" output="screen">
    <param name="NAO_IP" value="$(arg nao_ip1)"/>
    <param name="NAO_PORT" value="$(arg nao_port1)"/>
    <param name="H_MIN" value="0"/>
    <param name="H_MAX" value="10"/>
    <param name="S_MIN" value="160"/>
    <param name="S_MAX" value="255"/>
    <param name="V_MIN" value="160"/>
    <param name="V_MAX" value="255"/>
  </node>
  

  <!--
    Behavior Tree 1
  -->
  <node pkg="nao_behavior_tree" type="nao_behavior_tree" name="nao_behavior_tree1" output="screen">
    <param name="file" value="/home/olivier/ros_workspace/nao_behavior_tree/bt1_test_bearing.txt"/>
  </node>


  <!--
    Localization
  -->
  <node pkg="nao_behavior_tree" type="localization" name="localization" args="0" output="screen">
	<param name="camera" value="1"/>
	
    <param name="H_MIN" value="0"/>
    <param name="H_MAX" value="16"/>
    <param name="S_MIN" value="110"/>
    <param name="S_MAX" value="255"/>
    <param name="V_MIN" value="30"/>
    <param name="V_MAX" value="255"/>
  </node>
  
</launch>
